Adding the mook/wanderer mob for planetstation

Ergovisavi opened the pull request:
Adds another mob for planetstation, the mook/wanderer. They're low health mobs (45, which is three knife stabs) and intended to be fairly unthreatening while solo. In groups they become significantly harder to deal with as your opportunities to attack are diminished.

Designed to be an easier to read combatant where you can tell its states a bit better and shouldn't get meleed "at random". I'm definitely not making any concessions for player control because if you want their gameplay go play an ash lizard you donk.

They look like this:
![mook](https://user-images.githubusercontent.com/4465459/27843511-51fdd1d0-60c9-11e7-8dd1-fa055eb4da20.png)

* Also fixes a small bug with player controlled leapers where they couldn't fire sometimes.
* Also makes the leaper's projectile deal damage to simple mobs as a catch-all for things it can't stun

:cl:
add: Adds another planetstation mob, the "wanderer/mook" to the backend
fix: Fixes a problem with player controlled leapers where they occasionally can't fire
/:cl:

Ergovisavi committed:
Adding the mook/wanderer mob for planetstation

Ergovisavi committed:
Adds the two .ogg's for the mook mob

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +360,215 @@
+	icon_living = "mook"
+	icon_dead = "mook_dead"
+	pixel_x = -16
+	pixel_y = -16
```
I feel like this is weird since it means their feet are lower than the turf they're on; same for the leaper, actually

Ergovisavi replied:
Visually makes sense to me but maybe I have a fucked up sense of aesthetics. If people think that their pixel_y shouldn't be modified I don't have any qualms changing it to 0, I guess.

RemieRichards replied:
almost (actually no?) no mobs/objects extend below their turf.
Plenty extend NORTH, SOUTH and EAST out of their turf, but none below (SOUTH).

dannno commented:
cool ass sprite

RandomMarine commented:
If it wields a hatchet it should drop a hatchet.

Ergovisavi commented:
@RandomMarine This is just establishing the mob, like with the Leaper. All loot and stuff like that will come after the fact... if I ever finish up on these mobs.

PKPenguin321 commented:
these sprites are so good

it would be neat if these guys were their own race native to planetstation but that would probably definitely be too different from what's already going on here

Armhulen commented:
we should really replace ashwalkers with these cool looking guys

Armhulen commented:
also it's another i get really excited about new mobs episode

Repukan commented:
Why not add a variant of them to Steelpoint'caust ruin?

Repukan commented:
Due to crashed ship and shit, they're weak anyways, right?

WJohn commented:
Interesting that it has actual attack animations.

I'm not a huge fan of how it doesn't follow the 100% side view that mobs are supposed to have. Heck, even the front sprite isn't really facing the front, it's slightly at an angle (or maybe I'm looking at it wrong).

Cool sprite otherwise.

DoctoreMad commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +360,215 @@
+	return ..()
+
+/mob/living/simple_animal/death()
+	desc = "A deceased and unhealthy looking primitive. Upon closer inspection, it was suffering from severe cellular degeneration and its garments are machine made..."//Can you guess the twist
```
No I can't. I can't figure it out at all.

Ratvar replied:
autocloned descendants of past ships

Armhulen replied:
OH MY GOD IT'S THE SHELLS

Armhulen replied:
GASP

Iamgoofball replied:
why is this on the base simple animal

Ergovisavi replied:
Oops. Bad copy paste lol. Will fix.

dannno replied:
do you mean diseased instead of deceased?

DoctoreMad replied:
This is a desc only for when the mob dies.

XDTM replied:
time to reopen my shell pr, the lore is in

DoctoreMad commented:
Also I know ash liggers are shit but why would those replace them? They're kinda huge for a ghost spawn player character.

FantasticFwoosh commented:
Mooks are simple creatures, its a shit excuse to edge them out of the game when ashwalkers are superior because mooks are weaker.

> _Replace ashwalkers with mooks, suddenly lavaland ashwalker ruin is just a mook camp miners can clear out in 10 seconds_. Yeah unless you have a arguement that's a shitty and selfish reason to support them without actually giving a damn about planetstation.

If the consistent plan is to make a above & belowground, i can't see why not just keep ashwalkers as the below ground natives, and Mook's just as a annoying pest aboveground as the OP has provided with the commit detail.

Niknakflak commented:
That sprite looks so out of place and made for another game it's not even funny.  
Heck, was it even made for ss13 or is it torn out of someplace ergo?

Ergovisavi commented:
Like the leaper mob's sprites, I created it myself.

Ergovisavi committed:
mook fixes

Ergovisavi commented:
Removed the pixel_y offset on the leaper, changed the pixel_y offset on the mook to make it so it doesn't extend past the southern part of the tile it's on, fixed the description change on death pointing at all simple mobs

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+	ranged_cooldown_time = 10
+	pass_flags = LETPASSTHROW
+	robust_searching = 1
+	stat_attack = 1
```
I think this is a bool

Ergovisavi replied:
Nope. stat_attack = 0 means you only attack the awake, 1 includes unconscious, 2 includes dead.

RemieRichards replied:
Then you should use defines.
(Amusingly, these line up perfectly with CONSCIOUS, UNCONSCIOUS, DEAD)

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+	ranged = 1
+	ranged_cooldown_time = 10
+	pass_flags = LETPASSTHROW
+	robust_searching = 1
```
I think this is a bool

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+	icon_dead = "mook_dead"
+	pixel_x = -16
+	pixel_y = -8
+	ranged = 1
```
I think this is a bool

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+	if(istype(O, /mob/living/simple_animal/hostile/jungle/mook))
+		var/mob/living/simple_animal/hostile/jungle/mook/M = O
+		if(M.attack_state == MOOK_ATTACK_ACTIVE && M.throwing)
+			return 1
```
TRUE

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+			if(target)
+				if(isliving(target))
+					var/mob/living/L = target
+					if(L.incapacitated() && L.stat < 2)
```
stat != DEAD

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+		if(!struck_target_leap)
+			update_icons()
+		struck_target_leap = FALSE
+		if(prob(40) && !ckey)
```
use !client instead

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+		ranged_cooldown = world.time + ranged_cooldown_time
+		update_icons()
+		if(target && !stat)
+			icon_state = "mook"
```
put this in update_icon

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+/mob/living/simple_animal/hostile/jungle/mook/proc/ResetNeutral()
+	if(attack_state == MOOK_ATTACK_RECOVERY)
+		attack_state = MOOK_ATTACK_NEUTRAL
+		notransform = FALSE
```
You shouldn't be touching notransform

ChangelingRain replied:
that would be ideal; a better fix would be to make simple animals respect being stunned, and thus apply a stun that ignores their lack of CANSTUN flag

Ergovisavi replied:
What else do we have for "FUCK YOU, YOUR INPUTS DO NOT MATTER", that isn't also available to players somehow?

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+/mob/living/simple_animal/hostile/jungle/mook/proc/WarmupAttack(forced_slash_combo = FALSE)
+	if(attack_state == MOOK_ATTACK_NEUTRAL && target)
+		attack_state = MOOK_ATTACK_WARMUP
+		notransform = TRUE
```
You shouldn't be touching notransform

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,215 @@
+		var/mob/living/L = hit_atom
+		if(CanAttack(L))
+			L.attack_animal(src)
+			icon_state = "mook_strike"
```
put this in update_icon

Ergovisavi committed:
Mook fixes

Ergovisavi committed:
more fixes for mook

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -516,17 +514,32 @@
 		var/mob/living/L = hit_atom
 		if(CanAttack(L))
 			L.attack_animal(src)
-			icon_state = "mook_strike"
```
did you actually put an entry in update icons for this?

Ergovisavi replied:
I did. All the icons are working properly.

Ergovisavi committed:
Makes the mook's slash combo do an animation instead of just retaining the same pose

Ergovisavi commented:
This is ready whenever.

Cyberboss commented on code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm:
```diff
@@ -353,4 +359,230 @@
+			if(MOOK_ATTACK_WARMUP)
+				icon_state = "mook_warmup"
+			if(MOOK_ATTACK_ACTIVE)
+				if(density == FALSE)
```
if(!density)

Cyberboss committed:
reh
